app/src/main/java/ithaca_transit/android/cornellappdev/com/ithaca_transit/Adapters/SectionAdapter.java

```diff
@@ -69,39 +69,116 @@ public void onClick(View view) {
+        // Set direction section
+        holder2.direction_destination_one.setText(routeDirections.get(0).getName());
 
-        // Remove last element from route directions when we add icons, as we don't need icon for
-        // arrival
-        LinearLayoutManager iconsLayoutManager = new LinearLayoutManager(mContext,
-                LinearLayoutManager.VERTICAL, false);
-        holder2.icons.setLayoutManager((RecyclerView.LayoutManager) iconsLayoutManager);
-        holder2.icons.setAdapter(new IconsListAdapter(mContext, routeDirections));
+        // Want to set distance in middle of card if only walk route
+        if (size == 2 && routeDirections.get(0).getType().equals("walk") && routeDirections.get(
+                1).getType().equals("walk")) {
+            holder2.direction_destination_two.setText(routeDirections.get(0).getDistance() + " ft");
 
-        LinearLayoutManager dotsLayoutManager = new LinearLayoutManager(mContext,
-                LinearLayoutManager.VERTICAL, false);
-        holder2.dots.setLayoutManager((RecyclerView.LayoutManager) dotsLayoutManager);
-        holder2.dots.setAdapter(new DotsListAdapter(mContext, routeDirections));
+            holder2.direction_destination_two.setTextColor(Color.GRAY);
+            holder2.direction_destination_three.setText(routeDirections.get(1).getName());
+        } else {
+            holder2.direction_destination_two.setText(routeDirections.get(1).getName());
+
+            if (size >= 3) {
+                holder2.direction_destination_three.setText(routeDirections.get(2).getName());
+                if (routeDirections.get(2).getType().equals("walk")) {
+                    holder2.direction_distance_three.setText(
+                            routeDirections.get(2).getDistance() + " ft");
+
+                }
+            }
+            if (size >= 4) {
+                holder2.direction_destination_four.setText(routeDirections.get(size - 1).getName());
+                if (routeDirections.get(size - 1).getType().equals("walk")) {
+                    holder2.direction_distance_four.setText(
+                    routeDirections.get(size - 1).getDistance() + " ft");
+                }
+            }
+        }
+
+        // Set images
+        if (size == 2 && routeDirections.get(0).getType().equals("walk") && routeDirections.get(
+                1).getType().equals("walk")) {
+            holder2.icon_image_four.setVisibility(View.VISIBLE);
+        } else {
+            if (routeDirections.get(0).getType().equals("walk")) {
+                holder2.icon_image_one.setVisibility(View.VISIBLE);
+            } else {
+                holder2.container_bus_one.setVisibility(View.VISIBLE);
+                TextView number = holder2.container_bus_one.findViewById(R.id.tv_bus_number_one);
+                number.setText(routeDirections.get(0).getRouteNumber().toString());
+            }
 
+            if (size >= 3) {
```

Comment:
can you make sure that you include screenshots of all these cases, or as many as possible?